lib/Core/SQLiteBuildDB.cpp

```diff
@@ -626,25 +623,42 @@ class SQLiteBuildDB : public BuildDB {
     return DBKeyID(sqlite3_last_insert_rowid(db));
   }
 
-  KeyType getKeyForID(DBKeyID keyID) {
-    int result;
+  std::unordered_map<uint64_t, KeyID> engineKeyIDs;
 
-    // Seach for the key.
-    result = sqlite3_reset(findKeyNameForKeyIDStmt);
-    assert(result == SQLITE_OK);
-    result = sqlite3_clear_bindings(findKeyNameForKeyIDStmt);
-    assert(result == SQLITE_OK);
-    result = sqlite3_bind_int64(findKeyNameForKeyIDStmt, /*index=*/1, keyID.value);
-    assert(result == SQLITE_OK);
+  /// Maps a DBKeyID into an engine KeyID
```

Comment:
We should probably have a comment here about what expectations are around holding a lock around this call? I think we are effectively using `dbMutex` for this.

Reply:
Also, should we document somewhere clearly that the engine <-> db interface must keep KeyIDs stable across multiple builds? Otherwise we would need to reset that map on every build.